packages/cli/cloud/src/deploy-project/action.ts

```diff
@@ -0,0 +1,157 @@
+  name: string;
+  strapi?: {
+    uuid: string;
+  };
+};
+
+async function upload(ctx: CLIContext, project: ProjectInfos, token: string) {
+  const cloudApi = cloudApiFactory(token);
+  // * Upload project
+  try {
+    const storagePath = getTmpStoragePath();
+    const projectFolder = path.resolve(process.cwd());
+    const packageJson = (await loadPkg(ctx)) as PackageJson;
+
+    if (!packageJson) {
+      ctx.logger.error(
+        'Unable to deploy the project. Please make sure the package.json file is correctly formatted.'
+      );
+      return;
+    }
+
+    ctx.logger.log('📦 Compressing project...');
+    // hash packageJson.name to avoid conflicts
+    const hashname = crypto.createHash('sha512').update(packageJson.name).digest('hex');
+    const compressedFilename = `${hashname}.tar.gz`;
+    try {
+      ctx.logger.debug(
+        'Compression parameters\n',
+        `Storage path: ${storagePath}\n`,
+        `Project folder: ${projectFolder}\n`,
+        `Compressed filename: ${compressedFilename}`
+      );
+      await compressFilesToTar(storagePath, projectFolder, compressedFilename);
+      ctx.logger.log('📦 Project compressed successfully!');
+    } catch (error: unknown) {
+      ctx.logger.error(
+        '⚠️ Project compression failed. Try again later or check for large/incompatible files.'
+      );
+      ctx.logger.debug(error);
+      process.exit(1);
+    }
+
+    const tarFilePath = path.resolve(storagePath, compressedFilename);
+    const fileStats = fs.statSync(tarFilePath);
+
+    if (fileStats.size > FILE_SIZE_LIMIT) {
+      return ctx.logger.log(
+        'Unable to proceed: Your project is too big to be transferred, please use a git repo instead.'
+      );
+    }
+
+    ctx.logger.info('🚀 Uploading project...');
+    const progressBar = ctx.logger.progressBar(100, 'Upload Progress');
+
+    try {
+      const { data } = (await cloudApi.deploy(
+        { filePath: tarFilePath, project },
+        {
+          onUploadProgress(progressEvent) {
+            const total = progressEvent.total || fileStats.size;
+            const percentage = Math.round((progressEvent.loaded * 100) / total);
+            progressBar.update(percentage);
+          },
+        }
+      )) as { data: { buildId: string } };
+
+      progressBar.update(100);
+      progressBar.stop();
+      ctx.logger.success('✨ Upload finished!');
+      return data.buildId;
+    } catch (error: any) {
+      progressBar.stop();
+      if (error instanceof AxiosError && error.response?.data) {
+        if (error.response.status === 404) {
+          ctx.logger.error(
+            `The project does not exist. Remove the ${local.LOCAL_SAVE_FILENAME} file and try again.`
+          );
+        } else {
+          ctx.logger.error(error.response.data);
+        }
+      } else {
+        ctx.logger.error('An error occurred while deploying the project. Please try again later.');
+      }
+
+      ctx.logger.debug(JSON.stringify(error));
```

Comment:
You might want to use sth like https://github.com/sindresorhus/safe-stringify to be on the safe side on those error serializations.